### DISCUSSION TITLE
How to set PhpStorm as the Editor?

### DISCUSSION THREAD

**Discussion by BornSupercharged** [facelessuser/Rummage#471] [Q&A]
I tried using the path to PhpStorm but it doesn't do anything. Any help would be greatly appreciated!

**Comment by facelessuser** [comment]
I don't use PhpStorm. You'll need to figure out if and how PhpStrom can be fed a file from the command line with line context, then update the editor setting as outlined in the docs: https://facelessuser.github.io/Rummage/preferences/#editor

**Reply by BornSupercharged** [reply]
According to the PhpStorm docs:
https://www.jetbrains.com/help/phpstorm/opening-files-from-command-line.html#macos

```
Open a specific file on line number 42:
phpstorm --line 42 ~/MyProject/scripts/numbers.js
```

How would I format this properly in the "Editor" box, for compatibility with Rummage?

**Reply by facelessuser** [reply]
If `phpstorm` is in your path, then maybe this would work:

```
phpstorm --line {$line} "{$file}"
```

or

```
phpstorm.exe --line {$line} "{$file}"
```

If it can't find it on your path, you may need to use the full path:

```
"c:\path\to\phpstorm.exe" --line {$line} "{$file}"
```

I looked it up and you can specify column, so you could probably do this as well:

```
"c:\path\to\phpstorm.exe" --line {$line} --column {$col} "{$file}"
```

**Reply by BornSupercharged** [reply]
Thank you @facelessuser , on Mac 15.1.1 the correct Editor line is:
`phpstorm --line {$line} "{$file}"`

I appreciate your help!
